File: pt-1/3895.count-digit-appearances.java
/*
 * @lc app=leetcode id=3895 lang=java
 *
 * [3895] Count Digit Appearances
 *
 * Mirror the Python reference directly: convert each number to a string and
 * count occurrences of the requested digit.
 *
 * Time: O(total digits). Space: O(1) excluding temporary strings.
 */

// @lc code=start
class Solution {
    public int countDigitOccurrences(int[] nums, int digit) {
        char target = (char) ('0' + digit);
        int answer = 0;
        for (int number : nums) {
            String text = Integer.toString(number);
            for (int i = 0; i < text.length(); i++) {
                if (text.charAt(i) == target) {
                    answer++;
                }
            }
        }
        return answer;
    }
}
// @lc code=end

File: pt-1/3895.count-digit-appearances.go
package leetcode

//
// @lc app=leetcode id=3895 lang=golang
//
// [3895] Count Digit Appearances
//
// Notes
// Convert each number to decimal text and count occurrences of the requested
// digit, matching the Python implementation directly. Time is proportional to
// the total number of decimal digits. Space: O(1) besides conversion strings.
//
// @lc code=start

import (
	"strconv"
	"strings"
)

func CountDigitOccurrences3895(nums []int, digit int) int {
	target := strconv.Itoa(digit)
	total := 0
	for _, number := range nums {
		total += strings.Count(strconv.Itoa(number), target)
	}
	return total
}

// @lc code=end
